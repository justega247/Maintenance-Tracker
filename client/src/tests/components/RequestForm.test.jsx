import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedRequestForm, { RequestForm } from '../../components/RequestForm';
import mockData from '../__mocks__/mockData';

let wrapper;
let props;

const createMockStore = configureMockStore([thunk]);
const store = createMockStore({});
describe('RequestForm', () => {
  beforeEach(() => {
    props = {
      fetchRequest: jest.fn(),
      onSubmit: jest.fn(),
    };
    wrapper = shallow(<RequestForm {...props} />);
  });

  test('should render RequestForm correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should set title on input change', () => {
    const value = 'broken table';
    const name = 'title';
    wrapper.find('input').at(0).simulate('change', {
      target: { name, value },
    });
    expect(wrapper.state(name)).toBe(value);
  });

  test('should set type on input change', () => {
    const value = 'maintenance';
    const name = 'type';
    wrapper.find('select').at(0).simulate('change', {
      target: { name, value },
    });
    expect(wrapper.state(name)).toBe(value);
  });

  test('should set description on input change', () => {
    const value = 'broken tables are everywhere';
    const name = 'description';
    wrapper.find('textarea').at(0).simulate('change', {
      target: { name, value },
    });
    expect(wrapper.state(name)).toBe(value);
  });

  test('should call the submit props', () => {
    const { request } = mockData;
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.setState(request);
    const instance = wrapper.instance();
    instance.onSubmit(event);
    expect(wrapper).toMatchSnapshot();
  });

  test('should call the submit props', () => {
    const { request } = mockData;
    const data = {
      ...request,
      title: '',
      description: '',
    };
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.setState(data);
    const instance = wrapper.instance();
    instance.onSubmit(event);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render connected component', () => {
    wrapper = shallow(<ConnectedRequestForm store={store} {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
