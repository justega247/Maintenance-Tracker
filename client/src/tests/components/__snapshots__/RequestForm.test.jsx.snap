// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestForm should call the submit props 1`] = `
<div
  className="request__form"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        className="label request__label"
        htmlFor="type"
      >
        Type:
        <select
          className="request__type"
          name="type"
          onChange={[Function]}
          value="repairs"
        >
          <option
            key="repairs"
            value="repairs"
          >
            repairs
          </option>
          <option
            key="maintenance"
            value="maintenance"
          >
            maintenance
          </option>
        </select>
      </label>
    </div>
    <div>
      <label
        className="label request__label"
        htmlFor="title"
      >
        Title:
        <div>
          <input
            className="form__input"
            name="title"
            onChange={[Function]}
            placeholder="enter a title"
            type="text"
            value="Broken hedges"
          />
        </div>
      </label>
    </div>
    <div>
      <label
        className="label request__label"
        htmlFor="description"
      >
        Description:
        <div>
          <textarea
            className="form__input"
            name="description"
            onChange={[Function]}
            placeholder="enter a description"
            value="We will need to check and get the hedges fixed"
          />
        </div>
      </label>
    </div>
    <div>
      <Button
        className="button--big"
        text="Save Request"
        type="submit"
      />
    </div>
  </form>
</div>
`;

exports[`RequestForm should call the submit props 2`] = `
<div
  className="request__form"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        className="label request__label"
        htmlFor="type"
      >
        Type:
        <select
          className="request__type"
          name="type"
          onChange={[Function]}
          value="repairs"
        >
          <option
            key="repairs"
            value="repairs"
          >
            repairs
          </option>
          <option
            key="maintenance"
            value="maintenance"
          >
            maintenance
          </option>
        </select>
      </label>
    </div>
    <div>
      <label
        className="label request__label"
        htmlFor="title"
      >
        Title:
        <div>
          <input
            className="form__input"
            name="title"
            onChange={[Function]}
            placeholder="enter a title"
            type="text"
            value=""
          />
        </div>
      </label>
      <div
        className="form__error"
        key="Sorry, you have to specify a title for your request"
      >
        Sorry, you have to specify a title for your request
      </div>
    </div>
    <div>
      <label
        className="label request__label"
        htmlFor="description"
      >
        Description:
        <div>
          <textarea
            className="form__input"
            name="description"
            onChange={[Function]}
            placeholder="enter a description"
            value=""
          />
        </div>
      </label>
      <div
        className="form__error"
        key="Sorry, you have to specify a description for your request"
      >
        Sorry, you have to specify a description for your request
      </div>
    </div>
    <div>
      <Button
        className="button--big"
        text="Save Request"
        type="submit"
      />
    </div>
  </form>
</div>
`;

exports[`RequestForm should render RequestForm correctly 1`] = `
<div
  className="request__form"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        className="label request__label"
        htmlFor="type"
      >
        Type:
        <select
          className="request__type"
          name="type"
          onChange={[Function]}
          value="repairs"
        >
          <option
            key="repairs"
            value="repairs"
          >
            repairs
          </option>
          <option
            key="maintenance"
            value="maintenance"
          >
            maintenance
          </option>
        </select>
      </label>
    </div>
    <div>
      <label
        className="label request__label"
        htmlFor="title"
      >
        Title:
        <div>
          <input
            className="form__input"
            name="title"
            onChange={[Function]}
            placeholder="enter a title"
            type="text"
            value=""
          />
        </div>
      </label>
    </div>
    <div>
      <label
        className="label request__label"
        htmlFor="description"
      >
        Description:
        <div>
          <textarea
            className="form__input"
            name="description"
            onChange={[Function]}
            placeholder="enter a description"
            value=""
          />
        </div>
      </label>
    </div>
    <div>
      <Button
        className="button--big"
        text="Save Request"
        type="submit"
      />
    </div>
  </form>
</div>
`;

exports[`RequestForm should render connected component 1`] = `
<RequestForm
  fetchRequest={[Function]}
  onSubmit={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
